src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,321 @@
+    }
+
+    private static void saveFile() {
+        try {
+            new File(DIR_PATH).mkdir();
+            new File(FILE_PATH).createNewFile();
+            FileWriter fileWriter = new FileWriter(FILE_PATH);
+            String text = "";
+            for (Task task : tasks) {
+                if (task instanceof ToDo) {
+                    text += (task.isDone) ? "T|1|" : "T|0|";
+                    text += task.description + System.lineSeparator();
+                } else if (task instanceof Deadline) {
+                    text += (task.isDone) ? "D|1|" : "D|0|";
+                    text += task.description + "|" + ((Deadline) task).by + System.lineSeparator();
+                } else {
+                    text += (task.isDone) ? "E|1|" : "E|0|";
+                    text += task.description + "|" + ((Event) task).at + System.lineSeparator();
+                }
+            }
+            fileWriter.write(text);
+            fileWriter.close();
+        } catch (IOException exception) {
+            System.out.println("Accessing data failed.");
+        }
+    }
+
+    private static void loadFile() throws InvalidFileException{
+        File file = new File(FILE_PATH);
+        try {
+            Scanner fileScanner = new Scanner(file);
+            while (fileScanner.hasNext()) {
+                String taskInfo = fileScanner.nextLine();
+                char taskType = taskInfo.charAt(0);
+                boolean taskIsDone = taskInfo.charAt(2) == '1';
+                taskInfo = taskInfo.substring(4);
+                switch (taskType) {
+                case 'T':
+                    try {
+                        addToDo(taskInfo, false);
+                        tasks.get(tasks.size() - 1).isDone = taskIsDone;
+                    } catch (InvalidCommandException exception) {
+                        throw new InvalidFileException();
+                    }
+                    break;
+                case 'D':
+                    try {
+                        addDeadline(taskInfo.replace("|","/by"), false);
+                        tasks.get(tasks.size() - 1).isDone = taskIsDone;
+                    } catch (InvalidCommandException exception) {
+                        throw new InvalidFileException();
+                    }
+                    break;
+                case 'E':
+                    try {
+                        addEvent(taskInfo.replace("|", "/at"), false);
+                        tasks.get(tasks.size() - 1).isDone = taskIsDone;
+                    } catch (InvalidCommandException exception) {
+                        throw new InvalidFileException();
+                    }
+                }
+            }
+        } catch (FileNotFoundException exception) {
+            saveFile();
+            loadFile();
+        }
+
+    }
+
+    private static void showLogo() {
+        String logo = " ____        _        \n"
```

Comment:
Although not necessary, the `logo` here can be treated as `CONSTANT` variable, which you can further refactor out.